### PR TITLE
Adds a preference to make flashes darken your screen

### PR DESCRIPTION
**Pull request by Ryll-Ryll**
## About The Pull Request
A player who suffers from epileptic seizures recently asked if there was anything they could do to avoid the bright white flash effect that comes along with getting hit by a flash or flashbang, and this PR is my simple attempt at providing a way to avoid these sudden flashes. Obviously, there's still lots of other possible triggers, and no game can ever claim to be truly safe, but hopefully this helps people avoid one of the most common triggers.

https://streamable.com/qs50sc

This PR simply adds a pref that replaces the bright white screen with a totally black screen, and nothing else. As I don't really have any standard for whether this would be effective at preventing triggers or if there's a better way to do this, I'd appreciate feedback from someone who's more familiar with epilepsy for any ways this can be approved.

## Why It's Good For The Game
Adds more accessibility options

## Changelog
:cl: Ryll/Shaps
qol: Adds a new preference for photosensitive players in the OOC preferences menu to replace the bright white flash from things like flashes/flashbangs with a solid black screen.
/:cl:

### CODE REVIEW

**Comment by DAKKA-WAAAGH** [comment]
I like this just because of how dark my room usually is when I'm playing and all of a sudden I get blinded IRL by my screen.

Good PR!!

**Comment by Ryll-Ryll** [comment]
I've since been told that epilepsy is triggered by sudden changes in contrast, which changing the sudden white screen into a sudden black screen wouldn't really fix. Looking into some other ideas

**Comment by AlinaStarkova** [comment]
> 
> 
> I've since been told that epilepsy is triggered by sudden changes in contrast, which changing the sudden white screen into a sudden black screen wouldn't really fix. Looking into some other ideas

it wont be a full fix, but changing to black is still objectively better than to white as a temporary solution, just FYI. most of the time the screen on ss13 is kinda dark, so going to black has less of a contrast change as going to white, so this PR would only become an issue if your in a really bright area when you get flashed because then you go to bright to dark, the opposite of dark to white.

so if we cant get other options done in code this is still a partial fix which is better than no fix when it comes to potentially causing seizures

**Comment by Wayland-Smithy** [comment]
Even if the accessibility gains might not be the largest it's still really nice darkmode QOL

**Comment by godspeedfunnyman** [comment]
> I like this just because of how dark my room usually is when I'm playing and all of a sudden I get blinded IRL by my screen.

Immersion and realism

**Comment by AMonkeyThatCodes** [comment]
i'm guessing mothblocks made an exception to the prefs freeze because this is accesibility?

**Comment by CRITAWAKETS** [comment]
epilepsy more has to do with how much light is going into your eyes and stuff like that, so a black screen helps a lot

**Comment by Mothblocks** [comment]
> i'm guessing mothblocks made an exception to the prefs freeze because this is accesibility?

:+1:

**Comment by AlinaStarkova** [comment]
> 
> 
> epilepsy more has to do with how much light is going into your eyes and stuff like that, so a black screen helps a lot

right, the change in light levels specifically, the current white flash would also be fine if it was just slowed down to slowly get that bright, so a black screen is better than white since ss13 is usually dark, theres still a risk if your in a bright area when it goes dark. hence i said this is only technically a partial solution but still very very better than the current white screen

**Comment by coiax** [comment]
> 
> 
> I've since been told that epilepsy is triggered by sudden changes in contrast, which changing the sudden white screen into a sudden black screen wouldn't really fix. Looking into some other ideas

It is more work, and possibly out of scope, but what about the "flashbang" effect from various FPS games. The current screen "imprints" on your screen, and slowly fades out to the actual live screen.

**Comment by Mothblocks** [comment]
its not possible on byond

**Comment by Rohesie** [comment]
What is the state of this PR? Is it ready?

File: code/modules/mob/living/living_defense.dm [hunk 1]
```diff
@@ -406,6 +406,9 @@
 	if(HAS_TRAIT(src, TRAIT_NOFLASH))
 		return FALSE
 	if(get_eye_protection() < intensity && (affect_silicon || override_blindness_check || !is_blind()))
+		if(client?.prefs?.darkened_flash)
+			type = /atom/movable/screen/fullscreen/flash/black
```
Comment (Rohesie) [inline code/modules/mob/living/living_defense.dm:410]: The `type` parameter never gets overridden, right?
This is a bad pattern we've started using, filling proc calls with arguments like these.
If you could remove it and just make it a proc variable, that'd be good.
It would imply changing the arguments on the 6 proc definitions, but thankfully there's no need to change any of the proc calls.

**Comment by maxymax13** [comment]
why not like a quick fade to mid dark grey, around the color of walls and floor tiles averaged

**Comment by Cyberboss** [comment]
> 
> 
> its not possible on byond

vis_contents + getFlatIcon()?

**Comment by AlinaStarkova** [comment]
> 
> 
> > its not possible on byond
> 
> vis_contents + getFlatIcon()?

from what i understand, getFlatIcon() is extremely poor for performance, i cant imagine how bad it will be for an entire screen when flashed

**Comment by Mothblocks** [comment]
ok let me rephrase

its not possible in byond without getFlatIcon, which is extremely slow just for 4x4 camera photos, let alone your entire screen 😛

File: code/modules/mob/living/living_defense.dm (hunk 1 above)
Comment (Ryll-Ryll) [reply code/modules/mob/living/living_defense.dm]: It actually does get overridden for silicon's flash_act to use the static version, which seems reasonable to mme. I've tweaked the proc a little bit to make this a little less silly looking, but this lets us make sure we're overriding the flash type which I think is justified since it's an accessibility thing.

**Comment by Ryll-Ryll** [comment]
I'm just gonna give the go ahead on this PR to be merged. A solid black screen isn't ideal and there's still definitely the possibility that someone in a really light environment getting smacked with like 5 flashbangs one after another could still be jarring and potentially dangerous, but I've been told my several photosensitive players that at least a black screen will be more preferable  than a full white one.

Maybe in the future someone more skilled than I am with filters and overlays and such can implement a more thorough solution
